### DISCUSSION TITLE
Openssl Compilation and linking with Apache server

### DISCUSSION THREAD

**Discussion by vish190694** [openssl/openssl#23439] [Q&A] [answered]
Hello all,

I have compiled the openssl1.0.2l and generated libssl.a and libcrypto.a libs.
While I try to compile and link the apache server with these libs am facing the below errors during linking,Please let me know how to resolve this issue

/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)
/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)
/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)
/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)
/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)
/toolchain/cavium/OCTEON/tools-3.1.2_build_50/bin/../lib/gcc/mips64-octeon-linux-gnu/4.7.0/../../../../mips64-octeon-linux-gnu/bin/ld: ./srclib/openssl/OBJ/ENB/ohn/libssl.a(s3_srvr.o): Relocations in generic ELF (EM: 62)

collect2: error: ld returned 1 exit status
make: *** [OBJ/ENB/ohn/websvr.ohn] Error 1
./srclib/openssl/OBJ/ENB/ohn/libssl.a: could not read symbols: File in wrong format

**Comment by tom-cosgrove-arm** [comment, marked as answer]
Do please note that OpenSSL v1.0.2l was released in 2017, and [v1.0.2anything is no longer supported](https://www.openssl.org/policies/releasestrat.html) - [it went out of support in 2019](https://www.openssl.org/blog/blog/2018/05/18/new-lts/).

Can you try with a more recent version of OpenSSL? Something in the 3.x range?

**Reply by vish190694** [reply]
I have tried upgrading the openssl version but it was not possible in my system as we have old archs which might not be compatible.
mips64-octeon-linux-gnu is the toolchain used to compile now,even with openssl1.0.2u Im facing the similar kind of issue
